plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -147,54 +149,6 @@ private Storage createClient(final String clientName, final GoogleCloudStorageCl
         return storageOptionsBuilder.build().getService();
     }
 
-    /**
-     * Pins the TLS trust certificates and, more importantly, overrides connection
-     * URLs in the case of a custom endpoint setting because some connections don't
-     * fully honor this setting (bugs in the SDK). The default connection factory
-     * opens a new connection for each request. This is required for the storage
-     * instance to be thread-safe.
-     **/
-    private static HttpTransport createHttpTransport(final String endpoint) throws Exception {
-        final NetHttpTransport.Builder builder = new NetHttpTransport.Builder();
-        // requires java.lang.RuntimePermission "setFactory"
-        builder.trustCertificates(GoogleUtils.getCertificateTrustStore());
-        if (Strings.hasLength(endpoint)) {
-            final URL endpointUrl = URI.create(endpoint).toURL();
-            // it is crucial to open a connection for each URL (see {@code
-            // DefaultConnectionFactory#openConnection}) instead of reusing connections,
-            // because the storage instance has to be thread-safe as it is cached.
```

Comment:
Do you know if this is still a requirement with the SDK in version 1.77.0?

Reply:
I don't think so. The two linked bugs are resolved and there is no connection caching at all even in the current version. It creates a new `DefaultConnectionFactory` (which doesn't cache in the first place) with every new `NetHttpTransport.Builder` => since we call `super` anyway even if the underlying `DefaultConnectionFactory` were to cache it would just be per instance and not globally => all good I think/hope? :)

Reply:
Ok, I've been a bit confused by the comment in the code. I checked the previous and current version of DefaultConnectionFactory and I think we're all good too. Thanks!